src/rules/commentFormatRule.ts

```diff
@@ -173,19 +239,37 @@ function walk(ctx: Lint.WalkContext<Options>) {
         if (start === end) {
             return;
         }
-        const commentText = fullText.slice(start, end);
+        let commentText = fullText.slice(start, end);
         // whitelist //#region and //#endregion and JetBrains IDEs' "//noinspection ..."
         if (/^(?:#(?:end)?region|noinspection\s)/.test(commentText)) {
             return;
         }
 
+        if (ctx.options.noSpaceExceptions !== undefined) {
+            const match = ctx.options.noSpaceExceptions.exec(commentText);
+            if (match !== null) {
+                const lengthOfMatch = match[0].length;
+                commentText = commentText.substring(lengthOfMatch);
+                // Check optional colon after keyword
+                if (commentText[0] === ":") {
```

Comment:
If folks want to ignore colons, shouldn't they put that in the regex? What if we don't want it or prefer ` - ` instead?

Reply:
Like my comment pointed out: it is only optional. You can type for example _TODO_ or _TODO:_, the second will then trigger the if statement

Reply:
More strictly, what if someone explicitly doesn't want `:`s to be allowed here? Let's say their coding standards are to have something like:

```typescript
// TODO 55 - enable x y z
```

Allowing a `:` by default, without a way to turn it off, would be disruptive for that user.

It's a minor point, but if this goes in, it would be a breaking change to disable it, which would also be disruptive.

Reply:
OK I think I have to write some new test for this
Will programm it when I have more time

Do you want to have a regexp to do this or should I only remove the optional colon and then everything is open?

currently it is only there because you can not / should not place a space between the TODO and the colon

So if you write `// TODO 55 - ...` then the part after TODO is already the comment

Reply:
+1 for a regexp!

Reply:
So, after I've looked at what I implemented then ~4 months ago, I came back to why regex here makes no sense.

This feature makes it possible to use keywords like TODO without a space between the comment marks and the beginning of the actual comment.
Therefore, it makes the option synonymous only and only sense in combination with `"check-space"`.

The optional colon is a common syntax and is not in conflict with any ' - ' behind the TODO.

Therefore, the following things are possible:

```js
//TODO my comment
//TODO: my comment
//TODO - my comment
//TODO: - my comment (dont think that anyone use this form ^^)
//TODO 55 - my comment (the comment starts after the TODO with 55...)
//TODO: 55 - my comment
```


The only option I have ever seen that could be "problematic" would be the following:

```js
//TODO(abc): comment (abc is an abbreviation)
```

But that comes from the c ++ style guide: https://google.github.io/styleguide/cppguide.html#TODO_Comments
In the example, again there is a space between the comment and the //

So if someone wants to force a comment style, he should use the other rule option ignore-pattern and even that's not really correct.

I only see two options:

1. It is not part of tslint to do anything to enforce a style of commenting
2. Implement a new rule / feature and open a proposal

Reply:
Ok @Shinigami92, you've convinced me 😄 sounds good. The way it was before, with or without the `:`, seems fine.